### ISSUE TITLE
Provide more info on error for MetricServer

### ISSUE THREAD

**Issue by meikww** [derailed/popeye#86]
<img src="https://raw.githubusercontent.com/derailed/popeye/master/assets/popeye_feature.png" align="right" width="100" height="auto"/>

In our result we see an error for line MetricServer and we are not sure what it means<br/>

![image](https://user-images.githubusercontent.com/61747259/75795012-f61ced80-5d71-11ea-9bb4-35ffc37b815a.png)

We're running Popeye against an Openshift 3.11 environment.<br/>

**Comment by derailed** [comment]
@meikww This error indicates that popeye was unable to connect to the metrics-server on your cluster. If you are not running one,  popeye won't be able to gather metrics related sanitization which is not a deal as the report should still complete. Is this the case here?

**Comment by meikww** [comment]
Hi derailed, thanks a lot for the quick reply. Yes, the report still completed. I'll check with our ops team if we're running metrics on that cluster. Assuming we're running the metrics server: where/how is Popeye looking for the metrics server?

**Comment by meikww** [comment]
Hi derailed, hawkular seems to be running fine on our cluster. How does Popeye try to reach the Metrics-Server / which endpoints does it expect / look for? Alternatively: Can we configure anything in Popeye to make it look at the right place in our environment?

**Comment by derailed** [comment]
@meikww Thank you for your reply! I am not familiar with this project and it looks like it using their own metrics server. Popeye looks for the [standard](https://github.com/kubernetes-sigs/metrics-server) metrics-server running on port 4443.  I am not sure if hawkular uses the same apis to gather node/pod/container metrics. This is currently not configurable.

**Comment by meikww** [comment]
Hi @derailed. Thanks a get. I mixed it up: we are running Heapster. It's running on port 80, though. Can we at least configure the port number somewhere?

**Comment by derailed** [comment]
@meikww I don't think this is a port config issue. Heapster has been deprecated. Metrics-server is now the recommended way to gather cluster metrics. I thing later versions of openshift now support it...

**Comment by meikww** [comment]
Hi @derailed, thanks a lot. In the mean time I checked further. As per my understanding in our OpenShift 3.11 environment Hawkular is the actual metrics server, Heapster just collects the metrics from the nodes and exports them to Hawkular. So, we would need to communicate with Hawkular (it'S running on port 443 it seems) and yes we would need to figure out if the API is compatible. We'll check this. Will take us some time. I think we can close this ticket. Would you be ok if we contribute to Popeye to support Hawkular? Can you point us to a good starting point for this particular requirement?

**Comment by MikaelSmith** [comment]
This now appears to be a hard failure with Popeye v0.9.1:
```
 ___     ___ _____   _____                       D          .-'-.
| _ \___| _ \ __\ \ / / __|                       O     __| K    `\
|  _/ _ \  _/ _| \ V /| _|                         H   `-,-`--._   `\
|_| \___/_| |___| |_| |___|                       []  .->'  X     `|-'
  Biffs`em and Buffs`em!                            `=/ (__/_       /
                                                      \_,    `    _)
                                                         `----;  |


Boom! 💥 No metrics-server detected
```

**Comment by derailed** [comment]
@MikaelSmith Thank you for the heads up! #165. Sorry my bad ;(

**Comment by skycaptain** [comment]
Hey, I'm running into this issue with the lastest version running against our Talos cluster. metrics-server is installed from [home-operations/charts-mirror](https://github.com/home-operations/charts-mirror). With popeye, I get:

```
$ popeye -A -l error
...
GENERAL [redacted] (2025-07-25T20:01:58+02:00)
┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅
  · Connectivity...................................................................................✅
  · MetricServer...................................................................................💥
...
```

```
$ popeye version
 ___     ___ _____   _____                       K          .-'-.     
| _ \___| _ \ __\ \ / / __|                       8     __|      `\  
|  _/ _ \  _/ _| \ V /| _|                         s   `-,-`--._   `\
|_| \___/_| |___| |_| |___|                       []  .->'  a     `|-'
  Biffs`em and Buffs`em!                            `=/ (__/_       /  
                                                      \_,    `    _)  
                                                         `----;  |     
Version:   0.22.1
Commit:    35b554961dcdabfa7aba6bd070673c6c24c70884
Date:      2025-01-28T15:31:22Z
```
